### ISSUE TITLE
Improved country code in contact.

### ISSUE THREAD

**Issue by glauberssoutodossantos** [chatwoot/chatwoot#6657]
**Is your enhancement request related to a problem? Please describe.**

As a chatwoot agent, I would like to be able to save new contacts in a simpler way, without having to put the + and country code.

**Describe the solution you'd like**

It would be interesting if it had the flag of the client's country.

![image](https://user-images.githubusercontent.com/99494469/224731615-a620780e-f2ff-4e50-8ac0-9545006a9c31.png)

![image](https://user-images.githubusercontent.com/99494469/224731184-be53cdce-4162-4c06-b904-4371e5062345.png)

**Comment by pranavrajs** [comment]
@iamsivin If we allow them to select the county like this. We should not show +91 at the beginning.

**Comment by iamsivin** [comment]
Okay, will update with this suggestion.

**Comment by scmmishra** [comment]
@pranavrajs I think we could show the country dial code, just with a better UI, it seems like useful data, also flags are not instantly recognisable/searchable, codes are.

We don't have to specifically format the phone number, but we can include the dial code along with the flag I guess. @pranavrajs @iamsivin WDYT?

P.S. Mercury has a good design in this regard [https://dribbble.com/shots/18050792-Mercury-Phone-Input](https://dribbble.com/shots/18050792-Mercury-Phone-Input)

**Comment by iamsivin** [comment]
Yes, showing the country dial code near the phone number help to recognize the number and helps with the search experience. Currently, the search support searching with dial code/name/country code.

**Comment by scmmishra** [comment]
@iamsivin where are we sourcing the data from? Is it from the backend?

**Comment by iamsivin** [comment]
We have a file called \`countries.js\` its from there. Flag and dial code is missing there so i added it in the file and showing it in the UI.

**Comment by iamsivin** [comment]
CC: @pranavrajs

**Comment by pranavrajs** [comment]
@iamsivin I guess showing it in a masked format as the example in Mercury makes sense.
